pkg/cloudprovider/aws/block_store.go

```diff
@@ -42,7 +40,7 @@ const regionKey = "region"
 // from snapshot.
 var iopsVolumeTypes = sets.NewString("io1")
 
-type blockStore struct {
+type BlockStore struct {
```

Comment:
I notice that all the stucts for object and block store are now public, but I'm not sure I understand why. Is this required even though the types conform to the interfaces?

Reply:
Argh. Also: good catch!

Reply:
It's not technically required by go - you can return an unexported type from an exported function.  Linters usually recommend against doing this.  It can be annoying because if another package wants to store a reference to it, they either have to store it as an `interface{}` or as some interface that the unexported type implements.  And I think the only thing that not exporting the type buys you is that other packages can't directly instantiate an instance of the type. Either way, the fields of the type are unexported so they can't be accessed by other packages.  I'd probably choose exporting it over not exporting it, but we can go either way.

Reply:
Ok - I'm fine leaving it exported. Just wanted to see what the reasoning was.

Reply:
I'm sure it was a gut reaction on my part. I looked into this just now and still think it's weird to return an unexported type from an exported function. An interface would be better but exported type > interface? What makes this weird is that we are not exporting any element of that struct.